wrappers/services.go

```diff
@@ -562,6 +589,16 @@ func EnsureSnapServices(snaps map[*snap.Info]*SnapServiceOptions, opts *EnsureSn
 			// and if there are per-snap options specified, use that for
 			// VitalityRank
 			genServiceOpts.VitalityRank = snapSvcOpts.VitalityRank
+			genServiceOpts.QuotaGroup = snapSvcOpts.QuotaGroup
+
+			if snapSvcOpts.QuotaGroup != nil {
+				err := neededQuotaGrps.AddAllNecessaryGroups(snapSvcOpts.QuotaGroup)
```

Comment:
(nitpick^2) this could be a single line `if err := ... ; err != `

Reply:
this is addressed in https://github.com/snapcore/snapd/pull/10204